src/common/arrstr.cpp

```diff
@@ -573,3 +579,165 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
 
     return ret;
 }
+
+
+namespace
+{
+    enum wxStringFragmentType
+    {
+        wxFRAGMENT_TYPE_EMPTY = 0,
+        wxFRAGMENT_TYPE_ALPHA = 1,
+        wxFRAGMENT_TYPE_DIGIT = 2
+    };
+
+
+    // ----------------------------------------------------------------------------
+    // wxStringFragment
+    // ----------------------------------------------------------------------------
+    // 
+    // Lightweight object returned by wxString::GetNaturalFragment()
+    // Consists of either a string which represents a number, or a string which contains no numerical digits.
+    // Is used by wxString::CompareNatural()
+    class wxStringFragment
+    {
+    public:
+        wxStringFragment()
+            : type(wxFRAGMENT_TYPE_EMPTY)
+        {}
+
+        wxString             text;
+        double               value;
+        wxStringFragmentType type;
+    };
+
+
+    wxStringFragment GetFragment(wxString &text)
+    {
+        static wxRegEx naturalNumeric(wxS("[0-9]+"));
+        static wxRegEx naturalAlpha(wxS("[^0-9]+"));
+
+        size_t digitStart = 0;
+        size_t digitLength = 0;
+        size_t alphaStart = 0;
+        size_t alphaLength = 0;
+        wxStringFragment fragment;
+
+        if (text.empty())
```

Comment:
[wxWidgets Coding Guidelines](https://www.wxwidgets.org/develop/coding-guidelines/#spaces_keywords) require putting a space around an expression in parentheses, e.g., `if ( x )` instead of `if (x)`. I am not sure how it is actually enforced when adding a new code into an existing file, but it seems that this convention is already widely used in _arrstr.cpp_.